extension/js/common/core/attachment.ts

```diff
@@ -219,7 +219,7 @@ export class Attachment {
     } else {
       // && !Attachment.encryptedMsgNames.includes(this.name) -- already checked above
       const isAmbiguousAscFile = /\.asc$/.test(this.name); // ambiguous .asc name
-      const isAmbiguousNonameFile = !this.name || this.name === 'noname'; // may not even be OpenPGP related
+      const isAmbiguousNonameFile = !this.name || (this.name === 'noname' && this.type !== 'application/octet-stream'); // may not even be OpenPGP related
```

Comment:
I updated this particular test as a file named "noname" with the type of "application/octet-stream". It is challenging to determine the appropriate characteristics of an ambiguous "noname" file.